### PR TITLE
Add 18 new BPB tasks for basic_skills, gen, and science/medical categ…

### PR DESCRIPTION
**Pull request by kyleclo**
Adds 12 new evaluation tasks and an integration test script.                                                                                                       

6 New generative tasks implemented as BPB-only:                                                                                                                    
    - lambada_bpb_0shot - Word prediction                                                                                                                            
    - coqa_bpb_0shot - Conversational QA                                                                                                                             
    - drop_bpb_5shot - Discrete reasoning                                                                                                                            
    - jeopardy_bpb_5shot - Trivia
    - naturalqs_open_bpb_5shot - Open-domain QA
    - squad_bpb_5shot - Reading comprehension

6 New multiple-choice tasks implemented as MC/RC, so supports all metrics including Acc, BPB, CE Loss, etc:
    - lab_bench_dbqa_{mc|rc}_3shot - Lab database QA
    - lab_bench_protocolqa_{mc|rc}_3shot - Lab protocol QA
    - medmcqa_{mc|rc}_5shot - Medical multiple choice
    - medqa_en_{mc|rc}_5shot - Medical QA
    - qasper_yesno_{mc|rc}_5shot - Scientific yes/no
    - sciriff_yesno_{mc|rc}_5shot - Scientific yes/no

Note that some tasks (squad, drop, coqa, naturalqs_open) have multiple valid gold answers stored as lists. For example, from squad: `{"label": ["23 years", "23", "23"]}`

Had to adjust the test. When `label` is a list, all continuations are treated as valid gold answers for BPB computation. 

Another implementation from OE Eval Internal would be to treat only the first continuation as valid. Needs review.

Integration Test script. This is marked as slow and excluded from CI. Run manually: `pytest src/test/test_metric_integration.py -v`

Basically it mocks out logits and makes sure metrics are calculated correctly

### CODE REVIEW

File: src/olmo_eval/tasks.py [hunk 1]
```diff
@@ -2805,6 +2805,81 @@ def doc_to_label(self, doc) -> int:
...
+    ),
+    "naturalqs_bpb_5shot": (
+        OEEvalTask,
+        {"dataset_path": "naturalqs_open", "dataset_name": "bpb_5shot", "metric_type": "bpb"},
+    ),
+    "squad_bpb_5shot": (
+        OEEvalTask,
+        {"dataset_path": "squad", "dataset_name": "bpb_5shot", "metric_type": "bpb"},
+    ),
+    # Science/medical BPB (7 tasks)
```
Comment (davidheineman) [inline src/olmo_eval/tasks.py]: The below tasks are also RC tasks. Would it be better to define the RC task and then specify a BPB task type? Similar to other metrics.

**Review by epwalsh — APPROVED** [review]
LGTM, though @davidheineman should give the final review

**Comment by kyleclo** [comment]
@davidheineman tested on olmo-core:
```
[2026-02-03 14:37:04] INFO     [olmo_core.train.callbacks.evaluator_callback:160, rank=0] Finished sciriff_yesno_rc_5shot evals in 35.2 seconds. Metrics:
    sciriff_yesno_rc_5shot (accuracy)=0.2984
    sciriff_yesno_rc_5shot (accuracy v2)=0.2984
    sciriff_yesno_rc_5shot (CE loss)=4.576
    sciriff_yesno_rc_5shot (CE loss v2)=3.304
    sciriff_yesno_rc_5shot (BPB)=6.603
    sciriff_yesno_rc_5shot (BPB v2)=4.765
    sciriff_yesno_rc_5shot (soft loss)=0.3566
    sciriff_yesno_rc_5shot (soft loss v2)=0.3566
    sciriff_yesno_rc_5shot (log soft loss)=-1.49E+00
    sciriff_yesno_rc_5shot (log soft loss v2)=-1.49E+00
[2026-02-03 14:37:04] INFO     [olmo_core.train.callbacks.evaluator_callback:208, rank=0] Evaluation speed:
    lab_bench_protocolqa_rc_3shot (accuracy) (+variants): 0.8 sec (7 batches)
    naturalqs_bpb_5shot (BPB) (+variants): 1.1 sec (10 batches)
    jeopardy_bpb_5shot (BPB) (+variants): 3.4 sec (21 batches)
    qasper_yesno_rc_5shot (accuracy) (+variants): 4.2 sec (38 batches)
    squad_bpb_5shot (BPB) (+variants): 7.7 sec (72 batches)
    lab_bench_dbqa_rc_3shot (accuracy) (+variants): 7.8 sec (116 batches)
    lambada_bpb_0shot (BPB) (+variants): 9.1 sec (50 batches)
    drop_bpb_5shot (BPB) (+variants): 11.2 sec (77 batches)
    medmcqa_rc_5shot (accuracy) (+variants): 26.8 sec (243 batches)
    coqa_bpb_0shot (BPB) (+variants): 33.3 sec (444 batches)
    sciriff_yesno_rc_5shot (accuracy) (+variants): 35.2 sec (244 batches)
    medqa_en_rc_5shot (accuracy) (+variants): 43.0 sec (455 batches)
    Total evaluation time: 183.6 seconds (1777 batches)
[2026-02-03 14:37:04] WARNING  [olmo_core.train.trainer:592, rank=0] Run canceled from all ranks. Reason: canceled from evaluator callback since 'cancel_after_first_eval' is set
[2026-02-03 14:37:04] INFO     [olmo_core.train.callbacks.gpu_memory_monitor:41, rank=0] GPU capacity: NVIDIA H100 80GB HBM3 with 79.19GiB memory of which 17.04GiB is currently allocated and 17.39GiB is currently reserved.
[2026-02-03 14:37:04] INFO     [olmo_core.train.callbacks.speed_monitor:96, rank=0] Device: NVIDIA H100 80GB HBM3, Device peak FLOPS: 989500000000000
[2026-02-03 14:37:04] INFO     [olmo_core.train.callbacks.console_logger:67, rank=0] [step=0/75085,epoch=1]
    throughput/in-loop eval batches=1,777
    throughput/in-loop eval time (s)=183.6
    ```
